paxos/paxos.cc

```diff
@@ -51,6 +50,12 @@ PaxosResult Paxos::Propose(const std::string& value, uint8_t* leader_node_id,
     }
   }
 
+  // If the new leader is in the process of catching up, do not service
+  // reads/writes.
+  while (!this->paxos_node_->IsLeaderCaughtUp()) {
```

Comment:
Does this mean we're the leader here, or does this delay even happen on a replica?

Reply:
No, we should only reach here for leader